genienlp/models/general_seq2seq.py

```diff
@@ -129,17 +129,25 @@ def __init__(self, config=None, *inputs, **kwargs):
     def set_train_context_embeddings(self, trainable):
         self.encoder.set_train_context_embeddings(trainable)
 
-    def add_new_vocab_from_data(self, splits):
+    def add_new_vocab_from_data(self, splits, resize_decoder=False):
+        """
+        resize_decoder: if True, will actually resize the embedding matrix of the decoder
+        """
         #logger.info(f'Vocabulary has {self.numericalizer.num_tokens} tokens from training')
         old_num_tokens = self.numericalizer.num_tokens
         new_words = []
         for task_splits in splits:
             for split in task_splits:
+                if isinstance(split, Example):
+                    # single Example
+                    split = [split]
```

Comment:
This is really ugly. Can we make the callers pass the right amount of nested lists instead? We had to fix this code like two or three times for different paths...